### PR TITLE
create content fix - Yamlordereddict -> ryaml load + test infra

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/22093

## Description
using the ryaml load instead of the orderedloader
also added sanity unit test to the copy_dir_files in the content creator

### CODE REVIEW

**Comment by avidan-H** [comment]
I'd feel more comfortable with @yaakovi's review as he worked intimately on this stuff

File: demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import filecmp
+from demisto_sdk.commands.create_artifacts.content_creator import *
+
+
+def test_copy_dir_files():
+    current_dir = os.path.dirname(__file__)
+    content_repo_example = os.path.join(current_dir, '/test_files/content_repo_example')
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: you already using join...
```suggestion
    content_repo_example = os.path.join(current_dir, 'test_files', 'content_repo_example')
```

File: demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    # delete all files in the content_bundle
+    for filename in os.listdir(content_bundle_full_path):
+        file_path = os.path.join(content_bundle_full_path, filename)
+        try:
+            if os.path.isfile(file_path) or os.path.islink(file_path):
+                os.unlink(file_path)
+            elif os.path.isdir(file_path):
+                shutil.rmtree(file_path)
+        except Exception as err:
+            print('Failed to delete %s. Reason: %s' % (file_path, err))
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: should be in teardown as this is not part of the test.

File: demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import filecmp
+from demisto_sdk.commands.create_artifacts.content_creator import *
+
+
+def test_copy_dir_files():
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: test is unclear, what are you trying to test here?

File: demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import filecmp
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: please update file name: `content_creator_test.py`

File: demisto_sdk/commands/create_artifacts/content_creator.py [hunk 5]
```diff
@@ -169,17 +170,23 @@ def copy_dir_yml(self, dir_path, bundle):
...
 
             with open(path, 'r') as file_:
                 yml_info = yaml.safe_load(file_)
-
             ver = yml_info.get('fromversion', '0')
             print(f' - processing: {ver} ({path})')
-            copy_func(path, os.path.join(bundle, os.path.basename(path)), yml_info)
+
+            if dir_name in ['Playbooks', 'TestPlaybooks']:
+                self.copy_playbook_yml(path, os.path.join(bundle, os.path.basename(path)))
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/content_creator.py]: `TestPlaybooks` folder can contain scripts, make sure this is handled correctly

File: demisto_sdk/commands/create_artifacts/content_creator.py [hunk 6]
```diff
@@ -169,17 +170,23 @@ def copy_dir_yml(self, dir_path, bundle):
...
+            else:
+                ryaml = YAML()
+                ryaml.allow_duplicate_keys = True
+                with io.open(path, mode='r', encoding='utf-8') as file_:
+                    yml_unified = ryaml.load(file_)
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/content_creator.py]: no actual reason to open the file twice (here and on lines 177-178).
read it once using `ruamel` package.

extra: you can also modify the function `Unifier.write_yaml_with_docker` to use only the unified version of the dict so you don't have to pass it twice.
extra2: same goes for the script object, it can be calculated inside the function.

File: demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+from demisto_sdk.commands.create_artifacts.content_creator import *
+
+
+def test_copy_dir_files():
+    current_dir = os.path.dirname(__file__)
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: use absolute paths using `git_path()` (see other tests)

File: demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py (hunk 3 above)
Comment (anara123) [inline demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: `copy_dir_files` is internal details. 
instead test public functionality like `ContentCreator.run` function. 
And add test description, what is the behaviour that you are testing here

Comment (bakatzir) [reply demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: It's one path after the `create_content`, and I wanted to avoid the zipping on circle.
I agree we should do it, bur would preferred it in a 2nd batch, as this part have no tests at all.

File: demisto_sdk/commands/create_artifacts/content_creator.py (hunk 6 above)
Comment (bakatzir) [reply demisto_sdk/commands/create_artifacts/content_creator.py]: Done(I think)

File: demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py (hunk 3 above)
Comment (bakatzir) [reply demisto_sdk/commands/create_artifacts/tests/contet_creator_test.py]: added in the test description.

File: demisto_sdk/commands/create_artifacts/content_creator.py (hunk 5 above)
Comment (bakatzir) [reply demisto_sdk/commands/create_artifacts/content_creator.py]: You don't want to get the scripts in the Testplaybook folder to the content bundle afaik.
what do you want to handle aside from that?

Comment (yaakovi) [reply demisto_sdk/commands/create_artifacts/content_creator.py]: those scripts should go to the test_bundle as they are being used by the test playbooks

File: demisto_sdk/commands/create_artifacts/content_creator.py [hunk 8]
```diff
@@ -168,25 +167,24 @@ def copy_dir_yml(self, dir_path, bundle):
         :return: None
         """
         scan_files, _ = get_yml_paths_in_dir(dir_path, error_msg='')
+        print(scan_files)
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/content_creator.py]: remove

File: demisto_sdk/commands/create_artifacts/tests/content_creator_test.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+from tempfile import mkdtemp
+
+from demisto_sdk.commands.create_artifacts.content_creator import *
+from demisto_sdk.commands.common.git_tools import git_path
+
+
+class TestContentCreator:
+    def setup(self):
+        current_dir = os.path.normpath(os.path.join(__file__,
+                                                    f'{git_path()}/demisto_sdk/commands/create_artifacts', 'tests'))
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/tests/content_creator_test.py]: `git_path` returns absolute path to the root of the repo, no need to use the current file location.
```
current_dir = f'{git_path()}/demisto_sdk/commands/create_artifacts/tests'
```

File: demisto_sdk/commands/create_artifacts/tests/content_creator_test.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        - copying the content folder to a content bundle(flatten files)
+        Then
+        - ensure files are being flatten correctly
+        - ensure no !!omap are added(due to ordereddict yaml problems)
+        """
+        content_creator = ContentCreator(artifacts_path=self.content_repo, content_version='2.5.0',
+                                         content_bundle_path=self._dir,
+                                         preserve_bundles=False)
+        print(self.scripts_full_path)
+        print(self._dir)
```
Comment (yaakovi) [inline demisto_sdk/commands/create_artifacts/tests/content_creator_test.py]: don't forget to remove

File: demisto_sdk/commands/create_artifacts/content_creator.py (hunk 5 above)
Comment (bakatzir) [reply demisto_sdk/commands/create_artifacts/content_creator.py]: @yaakovi tended to it. first added the test case and fixed accordingly. hopefully I did not miss an edge case.
